services/ui-src/src/components/app/AppRoutes.tsx

```diff
@@ -1,10 +1,14 @@
 import { Navigate, Route, Routes } from "react-router-dom";
 import { Admin, Help, Home, NotFound, Profile } from "../../views";
+// components
+import { AdminBanner } from "components/banners/AdminBanner";
 // utils
-import { AdminBannerShape, UserRoles } from "utils/types/types";
+import { UserRoles } from "utils/types/types";
 
-export const AppRoutes = ({ adminBanner, userRole }: Props) => {
+export const AppRoutes = ({ userRole }: Props) => {
   const isAdmin = userRole === UserRoles.ADMIN;
+  const adminBanner = AdminBanner();
```

Comment:
Injecting the component down the stack seems odd to me in the scenario. What about taking the fetching and other state logic out of the component and placing it in a context instead? Then the context can be grabbed anywhere further down in the stack as needed, and the component itself becomes pure presentation.

Reply:
Yeah that's a good idea. I made that change, but then have been running into some weird issues related to [act()](https://reactjs.org/docs/test-utils.html#act) in the testing. Ideally I'd like to use context, but there's something above my head going on with the hooks that makes React upset. You can get the tests to pass locally but it still throws errors about it.

Reply:
Gonna hold off on useContext for now and re-evaluate later.